server/controllers/stock/reports/stock/entry/entryFromPurchase.js

```diff
@@ -0,0 +1,43 @@
+const db = require('../../../../../lib/db');
+const identifiers = require('../../../../../config/identifiers');
+
+const IS_EXIT = 0;
+const ENTRY_FROM_PURCHASE_ID = 1;
+
+/**
+ * @function fetch
+ * @description fetch stock entry from purchase
+ */
+function fetch(depotUuid, dateFrom, dateTo, showDetails) {
+  const sql = `
+  SELECT 
+    i.code, i.text, iu.text AS unit_text, BUID(m.document_uuid) AS document_uuid,
+    SUM(m.quantity) as quantity, m.date, m.description,
+    u.display_name AS user_display_name, sup.display_name AS supplier_display_name,
+    dm.text AS document_reference, d.text AS depot_name, 
+    CONCAT_WS('.', '${identifiers.PURCHASE_ORDER.key}', proj.abbr, p.reference) AS purchase_reference
```

Comment:
Can we use `dm.text` instead of this `CONCAT_WS`?   We noticed some bugs at HEV where the project wasn't correctly assigned in a few cases.

Reply:
No, i don't think so, because `CONCAT_WS` is for getting the purchase reference, there is no relation between purchase and stock movement, `dm.text` is the reference of the stock movement, `CONCAT_WS` is for the purchase reference

Reply:
Ah, what I mean is just do another JOIN on the `document_map` table on `p.uuid`.  Call it `dm2` or something.

If someone changes the project identifier, it will change this value.  We should probably keep this value static so that it always matches what is printed out on the receipts.